orm/models.py

```diff
@@ -241,6 +241,12 @@ def limit(self, limit_count: int):
         instance.pk = await self.database.execute(expr)
         return instance
 
+    async def get_or_create(self, **kwargs):
```

Comment:
Should we return a tuple of `(model_instance, was_created)` here? Some other ORM's (eg. [Django](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#get-or-create)) do this, with the use case being where the caller wants to perform additional work depending on whether anything actually changed.

Reply:
This is a good question. My personal opinion is that if you need additional logic in the scenario where the object was created, you probably shouldn't have used `get_or_create`.

However, people's familiarity with Django is a pretty strong argument for returning a tuple.

Reply:
Yeah, I understand your point (don't use a function that does X if you actually wanted to do Y). I think there are 2 distinct reasons why people use `get_or_create`:

1. Semantically, I simply want to ensure that a given object exists.
1. For clean code, I want to reuse a nontrivial block of code that forms part of a broader database update.

I personally find the Django `get_or_create` interface a bit clunky, so I don't feel committed to continuing that pattern in this library. A lot of the time I would be happy to use something simpler.

Just spitballing an alternative, what if we maintained the spirit and implementation of what you've done, but gave it a name that was more aligned with the purpose you've intended (like `ensure_exists` or some such)? That provides a clearer signal of how it's meant to be used, whilst leaving the door open for the more complex django-like version in the future, if the need arises. Further, that simpler use case probably justifies not supporting other complexities such as operator-containing fields, or a separate field list for creation.